### ISSUE TITLE
install.sh exits 1 after a fully successful install (signature + checksum verified, binary works)

### ISSUE THREAD

**Issue by Dicklesworthstone** [Dicklesworthstone/mcp_agent_mail_rust#327]
## Summary

`install.sh` completes a **fully successful install** — signature verified, checksum verified, binary installed and runnable — and then exits **1** with `Unexpected installer error`.

Found while running a real curl-bash end-to-end test against the published **v0.3.36** release in a sandboxed `HOME`.

## Everything substantive succeeds

From the verbose log:

```
binary-download:done       bytes=51304088
require_minisign:bin=/opt/homebrew/bin/minisign
sha256sums-download:done   bytes=666
minisig-download:done      bytes=330
verify_minisign:ok         manifest=.../SHA256SUMS
verify_checksum:ok         actual=2b90585878dd3c8509c21b177450f9f975caebb27a2f6a8601efed0ffb016523
maybe_add_path:appended guard to <HOME>/.zshenv
maybe_add_path:appended guard to <HOME>/.profile
maybe_add_path:done
update_mcp_configs:result  rc=0
update_mcp_configs:output  No coding agents detected. Use --agent to specify agents manually.
```

And the binary really is installed and really runs:

```console
$ <HOME>/.local/bin/am --version
am 0.3.36
```

## Then it reports failure

```
ERR Installer failed (exit 1) at line 1
ERR Unexpected installer error.
ERR Try re-running with --verbose for detailed diagnostics.
ERR If this persists, report at https://github.com/Dicklesworthstone/mcp_agent_mail_rust/issues
```

**The verbose log ends at `update_mcp_configs:output`.** Nothing after it is logged, so the non-zero exit originates *after* the last instrumented step, in code that does not log. `at line 1` is a bogus line number, which is consistent with an `ERR`/`EXIT` trap firing with a reset `LINENO`.

**Verified:** install succeeded; binary runs; exit code is 1; log ends with `update_mcp_configs:result rc=0`.
**Inferred, not proven:** the failure lives in a post-install trap or cleanup path — possibly on the "no coding agents detected" branch, since a sandboxed `HOME` has none.

## Why it matters

- `curl … | bash` inside a `set -e` script **aborts the caller** after a successful install.
- CI and automation that check the exit code see a failure that did not occur.
- Users get "Unexpected installer error" and an invitation to file a bug immediately after a working install.
- Tooling that gates on installer success can mark a healthy install as broken.

## Reproduce

```sh
T=$(mktemp -d /private/tmp/am-e2e-XXXXXX)   # NOT /tmp on macOS — see note below
curl -fsSL https://raw.githubusercontent.com/Dicklesworthstone/mcp_agent_mail_rust/main/install.sh -o "$T/install.sh"
env HOME="$T/home" bash "$T/install.sh"; echo "exit=$?"
"$T/home/.local/bin/am" --version           # works, despite exit=1
```

## Secondary note: macOS `/tmp` is refused (this part is correct behaviour)

Installing to any path under `/tmp` on macOS is refused:

```
!! install destination component is a symlink; refusing to write through it: /tmp
ERR Install destination contains traversal, a symlink, or a non-directory component
```

`/tmp` is a symlink to `/private/tmp` on macOS. **The anti-traversal hardening is correct and it fails safely and loudly** — no change requested. The only suggestion is that the message could mention that the resolved path (`/private/tmp/...`) is the usable one, since this is easy to hit when testing.

Environment: macOS 26.2, arm64, minisign at `/opt/homebrew/bin/minisign`, installer sha256 `14ee82c2292fddcf2807f27ba1c3198ad78ed36ba7d0e58fb944dcdb1f535e5d` (376,283 bytes).

**Comment by Dicklesworthstone** [comment]
I confirmed that the reported installer SHA-256 matches current main. The no-agent orchestration branch returns success in a focused test under macOS Bash 3.2, so I have not yet reproduced the full installer failure.

Could you share the complete stdout/stderr between `update_mcp_configs:output` and the final error, with bearer tokens redacted? Please also include the output of `command -v codex; command -v omp` from the same shell.

One distinction I am checking: the installer discovers remote clients through PATH as well as config directories, while native setup uses its own discovery. A detected remote client whose setup did not persist its credential intentionally causes installation to fail. I want to establish whether that mismatch occurred here before changing the exit-status behavior.

**Comment by Dicklesworthstone** [comment]
Two independently reproduced Bash 3.2 problems are now fixed in 4d181f8a:

- A fatal `set -u` error could be turned into exit 0 by Bash 3.2's EXIT trap handling. The installer now requires an explicit successful completion before returning success.
- Expected GNU/BSD `stat` fallbacks and Git non-repository probes could invoke inherited ERR handlers inside command substitutions. Those probes now preserve their failure status without printing a spurious installer failure.

Validation passed on Linux and native macOS 26.2 arm64 with stock Bash 3.2.57: 22 exit-status controls plus complete file and stdin installations of the real signed v0.3.36 binaries. Signature/checksum verification, exact installed versions, and absence of unexpected ERR diagnostics all passed. Existing artifact/fallback tests and 33 Git-authority assertions also passed. The ACFS installer checksum has been refreshed.

The full clean-HOME macOS control using your exact installer hash also exited 0, so these findings do not yet prove the cause of your original exit 1. Leaving this issue open pending the complete preceding stderr/PATH information requested above. Required-client setup failures still return nonzero; no credential or path-safety checks were relaxed.
